mineinabyss-systems/src/main/kotlin/com/mineinabyss/pvp/survival/SurvivalPvpFeature.kt

```diff
@@ -25,27 +34,46 @@ class SurvivalPvpFeature : AbyssFeature {
         )
 
         commands {
-            "pvp" {
-                "on" {
-                    playerAction {
-                        val data = player.playerData
-                        if (player.location.layer?.hasPvPDefault == false) {
-                            data.pvpStatus = true
-                            data.pvpUndecided = false
-                            data.showPvPMessage = true
-                            player.success("PvP has been enabled.")
-                        } else sender.error("PvP can not be toggled in this layer.")
-                    }
-                }
-                "off" {
-                    playerAction {
-                        val data = player.playerData
-                        if (player.location.layer?.hasPvPDefault == false) {
-                            data.pvpStatus = false
-                            data.pvpUndecided = false
-                            data.showPvPMessage = true
-                            player.success("PvP has been disabled.")
-                        } else sender.error("PvP can not be toggled in this layer.")
+            ("mineinabyss" / "mia") {
+                "pvp"(desc = "Commands to toggle pvp status") {
+                    permission = "mineinabyss.pvp"
+                    action {
+                        val player = sender as? Player ?: return@action
+                        if (player.location.layer?.hasPvpDefault == true) {
+                            player.error("Pvp cannot be toggled in this layer.")
+                            return@action
+                        }
+                        guiy {
+                            Chest(listOf(player), NegativeSpace.of(18) + "${ChatColor.WHITE}:pvp_menu:",
```

Comment:
Just add some parameters to the existing composable function (ex a boolean to toggle whether or not the prompt toggle should be shown) and reuse it.

Reply:
Not entirely sure what you mean by this?